internal/provisioner/plain/main.go

```diff
@@ -139,9 +142,18 @@ func main() {
 		RootDirectory: storageDirectory,
 		URL:           *storageURL,
 	}
+
+	var rootCAs *x509.CertPool
+	if bundleCAFile != "" {
+		var err error
+		if rootCAs, err = util.LoadCertPool(bundleCAFile); err != nil {
+			setupLog.Error(err, "unable to parse bundle certificate authority file")
+			os.Exit(1)
+		}
+	}
+
 	httpLoader := storage.NewHTTP(
-		// TODO: Use a trusted CA to connect to bundle content URLs.
-		storage.WithInsecureSkipVerify(true),
+		storage.WithRootCAs(rootCAs),
```

Comment:
Should we still expose a knob for using insecureSkipVerify, e.g. --insecure-skip-tls-verify?

Reply:
I'm not opposed, but do we have a use case where we need that flag now that this PR boostraps certs and CAs?

Reply:
I'd guess the use case would be dev environments, but if we're already leaning on cert-mgr to give us certificates for free we might be all set?